### ISSUE TITLE
Memory leak

### ISSUE THREAD

**Issue by iOSVlady** [ytmdesktop/ytmdesktop#699]
- [x] I understand that **YTMDesktop NO affiliation with Google or YouTube**.
- [x] I verified that there is no open issue for the same subject.

**Describe the bug**
Memory leak

**To Reproduce**
Listening songs for about 4 hours:)

**Expected behavior**
Less memory usage

**Screenshots**
![image](https://user-images.githubusercontent.com/46273878/115082309-64892480-9f0e-11eb-85f5-5e5f097ff7a3.png)

**Environment (please complete the following information):**
 * YTMDesktop version: ***latest***
 * OS: Windows

**Comment by Alipoodle** [comment]
This issue has been reported a few times over on the Discord server and it is currently believed to be an issue coming from YouTube Music and not the application.
However more investigation needs to go into the issue to see if this is preventable in some way.

**Comment by eegeeZA** [comment]
![image](https://user-images.githubusercontent.com/2684652/118087140-6509c380-b3c5-11eb-8708-329ad12c03cf.png)
Before and after force closing YTMDesktop.

**Comment by ckulas** [comment]
Seeing the same on mac 11.3.1 with Version 1.14:

<img width="771" alt="Bildschirmfoto 2021-05-15 um 19 16 27" src="https://user-images.githubusercontent.com/12587860/118372572-a7442800-b5b2-11eb-82cb-f78fe2dd18e7.png">

**Comment by sadortun** [comment]
Same on 1.13   on Win 10
![image](https://user-images.githubusercontent.com/794449/128610277-2611210d-0715-4004-9e16-9dd642ccc076.png)

**Comment by colin-creighton** [comment]
Still an issue for me

**Comment by Smity** [comment]
> This issue has been reported a few times over on the Discord server and it is currently believed to be an issue coming from YouTube Music and not the application. However more investigation needs to go into the issue to see if this is preventable in some way.

Not sure if this information will be helpful, but I have only experienced this problem when using the application. If I use YouTube Music in Chrome, the issue never presents. If I use the application the issue _always_ presents.

**Comment by cwrau** [comment]
Same here, I've set a memory limit of 1G but the application *really* tries not to use too much if it gets close and just comes to a grinding halt until it finally does use more than 1G and systemd kills and restart it.

It would be helpful, for me at least, if the app would just crash as fast as possible and maybe could just pick up where it left off?

**Comment by Alipoodle** [comment]
Continue where I left off is a feature, while not perfect has got improvements in v1.14 & lots of improvements in v2.
Understandable you might not get perfect return if you crash the application 😅 

Some people have noticed issues with leaving some integrations on like Discord Rich Presences when there's not a client to connect to.
However we've otherwise come to the conclusion that this is primarily YTM's fault with how it handles video buffer. 
TL;DR, YouTube music keeps all the data for the entire queue you have. You leave it running, it will keep the buffer of that first song you played, leading for it to stack up HEAVILY if you go song skipping.

**Comment by cwrau** [comment]
> Continue where I left off is a feature, while not perfect has got improvements in v1.14 & lots of improvements in v2.
> Understandable you might not get perfect return if you crash the application 😅 
> 
> Some people have noticed issues with leaving some integrations on like Discord Rich Presences when there's not a client to connect to.
> However we've otherwise come to the conclusion that this is primarily YTM's fault with how it handles video buffer. 
> TL;DR, YouTube music keeps all the data for the entire queue you have. You leave it running, it will keep the buffer of that first song you played, leading for it to stack up HEAVILY if you go song skipping.

Mh, I understand 🤔

Maybe there could be some form of internal checking of memory usage?
And if too much do a clean restart after the current song? Provided the "continue where left off" feature works

And crashing if using too much memory would help me a lot for now.
I understand that crashing can't necessarily lead to a good return, but coming to a grinding halt as it does now just leads me to have to kill it by hand and leaves me with an empty plate anyways.

And, tbh, an application should either just don't use too much memory (not always possible, I know) or just crash 😁

There shouldn't be a "I can't work without more memory but I'm crippling myself (and don't work correctly) just so I can live a couple more seconds until I eventually die because of OOM". That's helping no one 😅

**Comment by Alipoodle** [comment]
Closing issue due to no longer supportinng YouTube Music Desktop v1 + Release of YouTube Music v2.
Please re-open this issue if this is still occuring in the latest build of the application, however as a re-write has been done on the entire project most issues will be fixed, no longer a feature or re-worked.

**Comment by cwrau** [comment]
> Closing issue due to no longer supportinng YouTube Music Desktop v1 + Release of YouTube Music v2. Please re-open this issue if this is still occuring in the latest build of the application, however as a re-write has been done on the entire project most issues will be fixed, no longer a feature or re-worked.

Please reopen, as this still happens;

> Memory: 959.3M (max: 1.0G available: 64.6M peak: 1.0G)

Or is there a recommend amount of memory this is expected to consume?
